### PR TITLE
[FX-5263] Prepare Button migration

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-5263]

### Description

This PR contains all changes after follow up work before merging a feature branch to `master`.

### TODO

- [x] Add comment as we decided [here](https://github.com/toptal/picasso/pull/4265#discussion_r1595320264)
- [x] Remove tailwindcss from peerDepenencies

### How to test

- [Temploy](https://picasso.toptal.net/fx-5263-merge-feature-branch)
- Tested in SP and CP

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [x] test alpha package of Picasso in StaffPortal (https://github.com/toptal/staff-portal/pull/12589)

- [x] test alpha package of Picasso in ClientPortal (https://github.com/toptal/client-portal/pull/8951)

Checked in CP: https://github.com/toptal/client-portal/actions/runs/9017685046/job/24778165750?pr=8951
All broken tests are not related to `Button`

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5263]: https://toptal-core.atlassian.net/browse/FX-5263?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

File: .changeset/kind-shoes-push.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-shared': major
+---
+
+- remove addClassesToChildren helper
```
Comment (dmaklygin) [inline .changeset/kind-shoes-push.md:5]: Delete this file at all

File: packages/picasso-rich-text-editor/package.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "@toptal/picasso-shared": "^14.0.0",
     "react": ">=16.12.0 < 19.0.0",
     "react-dom": ">=16.12.0 < 19.0.0",
-    "typescript": "~4.7.0"
+    "typescript": "~4.7.0",
+    "tailwindcss": ">=3"
```
Comment (dmaklygin) [inline packages/picasso-rich-text-editor/package.json:36]: replace with "@toptal/picasso-tailwind" package

File: packages/base/Button/src/ButtonBase/ButtonBase.tsx [hunk 3]
```diff
@@ -110,6 +111,8 @@ export const ButtonBase: OverridableComponent<Props> = forwardRef<
 
   const titleCase = useTitleCase(propsTitleCase)
   const finalChildren = [titleCase ? toTitleCase(children) : children]
+  const rootElementName =
+    as !== 'button' && ('href' in props || 'to' in props) ? 'a' : undefined
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply packages/base/Button/src/ButtonBase/ButtonBase.tsx:115]: Add comment about adding this workaround.

This is to handle case when "as " is a Link, however, href and to are empty strings.
```
<Button
  as={Link}
  href=""
  to=""
/>

```

File: packages/base/Button/src/ButtonBase/test.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      })
+    })
+
+    describe('when "as" prop equals "Link" component', () => {
+      it('renders Button as a', () => {
+        const onClick = jest.fn()
+        const { container, getByText } = renderButton({
+          children: 'Click me!',
+          as: Link,
+          href: 'URL',
```
Comment (dmaklygin) [inline packages/base/Button/src/ButtonBase/test.tsx:79]: add case when "href" is an empty!

File: packages/base/Helpbox/package.json [hunk 5]
```diff
@@ -35,7 +35,8 @@
   ],
   "peerDependencies": {
     "@material-ui/core": "4.12.4",
-    "react": ">=16.12.0 < 19.0.0"
+    "react": ">=16.12.0 < 19.0.0",
+    "tailwindcss": ">=3"
```
Comment (dmaklygin) [inline packages/base/Helpbox/package.json:39]: replace with picasso-tailwind

File: packages/base/Modal/package.json [hunk 6]
```diff
@@ -37,7 +37,8 @@
   "peerDependencies": {
     "@material-ui/core": "4.12.4",
     "@toptal/picasso-provider": "*",
-    "react": ">=16.12.0 < 19.0.0"
+    "react": ">=16.12.0 < 19.0.0",
+    "tailwindcss": ">=3"
```
Comment (dmaklygin) [inline packages/base/Modal/package.json:41]: replace with picasso-tailwind

File: packages/base/Page/package.json [hunk 7]
```diff
@@ -39,7 +39,8 @@
     "@toptal/picasso-typography": "2.0.0",
     "@toptal/picasso-user-badge": "2.0.1",
     "@toptal/picasso-utils": "1.0.2",
-    "classnames": "^2.5.1"
+    "classnames": "^2.5.1",
+    "tailwind-merge": "^2.2.2"
```
Comment (dmaklygin) [inline packages/base/Page/package.json:43]: replace inPeerDependency: tailwindcss with picasso-tailwind

File: packages/base/Pagination/package.json [hunk 8]
```diff
@@ -26,15 +26,17 @@
...
   },
   "sideEffects": [
     "**/styles.ts",
     "**/styles.js"
   ],
   "peerDependencies": {
     "@material-ui/core": "4.12.4",
-    "react": ">=16.12.0 < 19.0.0"
+    "react": ">=16.12.0 < 19.0.0",
+    "tailwindcss": ">=3"
```
Comment (dmaklygin) [inline packages/base/Pagination/package.json:39]: replace with picasso-tailwind

File: packages/base/PasswordInput/package.json [hunk 9]
```diff
@@ -36,7 +36,8 @@
   ],
   "peerDependencies": {
     "@material-ui/core": "4.12.4",
-    "react": ">=16.12.0 < 19.0.0"
+    "react": ">=16.12.0 < 19.0.0",
+    "tailwindcss": ">=3"
```
Comment (dmaklygin) [inline packages/base/PasswordInput/package.json:40]: replace with picasso-tailwind

File: packages/base/PasswordInput/src/PasswordInput/PasswordInput.tsx [hunk 10]
```diff
@@ -80,7 +80,7 @@ export const PasswordInput = forwardRef<HTMLInputElement, Props>(
     const endAdornment = (
       <InputAdornment position='end'>
         <ButtonCircular
-          className={classes.toggle}
```
Comment (dmaklygin) [inline packages/base/PasswordInput/src/PasswordInput/PasswordInput.tsx:83]: remove toggle from styles

**Review by sashuk — APPROVED** [review]
Reviewed on MOB review

**Review by TomasSlama — APPROVED** [review]

**Pull request merged**